src/context.rs

```diff
@@ -279,8 +295,34 @@ impl<A: Actor> Context<A> {
                     histogram.start_timer()
                 };
 
+                #[cfg(not(feature = "timing"))]
                 msg.handle(actor, self).await;
 
+                #[cfg(feature = "timing")]
+                {
+                    match self.handler_timeout {
+                        None => msg.handle(actor, self).await,
+                        Some(timeout) => {
+                            let actor_type = std::any::type_name::<A>();
+                            let msg_type = msg.name();
```

Comment:
`actor_type` and `msg_type` are only used if `tracing` is enabled. I think you can move it down.